pkg/virt-controller/watch/vmi.go

```diff
@@ -729,6 +745,17 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		if !vmi.IsUnprocessed() {
 			return nil
 		}
+		// let's check if we already have topology hints or if we are still waiting for them
+		if vmi.Status.TopologyHints == nil && c.topologyHinter.TopologyHintsRequiredForVMI(vmi) {
+			if topologyHints, err := c.topologyHinter.TopologyHintsForVMI(vmi); err != nil {
+				c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedGatherhingClusterTopologyHints, err.Error())
+				return &syncErrorImpl{err, FailedGatherhingClusterTopologyHints}
+			} else if topologyHints != nil {
```

Comment:
do you still need this logic here that actually generates the hints for the VMI now that the code checks if the toplogy hints are nil && required? 

I think this is still causing multiple Events to get fired every reconcile loop. One in sync and another in the status update.

Reply:
Yes the purpose was to ensure that if for some reason a pod can't be created for some time that the hints can still be updated. Will remove it. It is probably over-optimized.